### PR TITLE
auto detect os if there is no preference

### LINKED ISSUE #401: OS specific instructions should detect current OS
When reading documentation that is specific depending on the operating system used, the page should try to determine the OS and show the appropriate tab by default.

Currently, the first tab is always shown regardless of the OS:

image

### PR DESCRIPTION
**Pull request by karthik2804**
Auto detects the OS if the user has not set a preference. 

fixes #401

### CODE REVIEW

File: static/js/src/modules/multiTab.js [hunk 1]
```diff
@@ -68,48 +70,65 @@ class multiTabContentHandler {
...
         }
-        if(updateLocalStorage) localStorage.setItem("toggleTabSelections", JSON.stringify(this.selectedTab))
+        if (updateLocalStorage) localStorage.setItem("toggleTabSelections", JSON.stringify(this.selectedTab))
+    }
+}
+
+function detectOS() {
+    let OS = null
+    let userAgent = navigator.userAgent.toLowerCase()
+    switch (true) {
```
Comment (itowlson) [inline static/js/src/modules/multiTab.js:120]: I haven't seen this idiom and it looks really strange to me but if it works it works - TIL!

**Review by itowlson — APPROVED** [review]

File: static/js/src/modules/multiTab.js (hunk 1 above)
Comment (karthik2804) [reply static/js/src/modules/multiTab.js:120]: I am a bit lost as to what you mean here.

Comment (itowlson) [reply static/js/src/modules/multiTab.js:120]: No action, I am just expressing that I learned something new from this PR.

Comment (karthik2804) [reply static/js/src/modules/multiTab.js:120]: ah!

Comment (karthik2804) [reply static/js/src/modules/multiTab.js:120]: Could have written it with if statements too!

**Pull request merged**